src/shared/Validation.js

```diff
@@ -25,9 +25,80 @@ export const validateOnInput = (
   const formatErrors = sectionFormatErrors.slice();
 
   switch (fieldType) {
-    case 'number': {
+    case 'int16': {
       const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
-      if (input && !validator.isInt(input)) {
+      const validInt = validator.isInt(input);
+
+      let validIntType = true;
+      if (parseInt(input, 10) < -32768 || parseInt(input, 10) > 32767) {
+        validIntType = false;
+      }
+
+      if ((input && !validInt) || (input && !validIntType)) {
+        inputValid = false;
+        if (idx === -1) {
+          formatErrors.push(FORM_ERRORS.INVALID_FORMAT);
+        }
+      }
+      else {
+        inputValid = true;
+        if (idx !== -1) {
+          formatErrors.splice(idx);
+        }
+      }
+      break;
+    }
+    case 'int64': {
+      const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
+      const validInt = validator.isInt(input);
+
+      let validIntType = true;
+      if (
+        parseInt(input, 10) < -9223372036854775808
+          || parseInt(input, 10) > 9223372036854775807
+      ) {
+        validIntType = false;
+      }
+
+      if ((input && !validInt) || (input && !validIntType)) {
+        inputValid = false;
+        if (idx === -1) {
+          formatErrors.push(FORM_ERRORS.INVALID_FORMAT);
+        }
+      }
+      else {
+        inputValid = true;
+        if (idx !== -1) {
+          formatErrors.splice(idx);
+        }
+      }
+      break;
+    }
+    case 'alphanumeric': {
+      const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
+      const valid = validator.isAlphanumeric(input);
+
+      if (input && !valid) {
+        inputValid = false;
+        if (idx === -1) {
+          formatErrors.push(FORM_ERRORS.INVALID_FORMAT);
+        }
+      }
+      else {
+        inputValid = true;
+        if (idx !== -1) {
+          formatErrors.splice(idx);
+        }
+      }
+      break;
+    }
+    case 'date': {
+      const idx = formatErrors.indexOf(FORM_ERRORS.INVALID_FORMAT);
+      const currentDate = (new Date()).toISOString();
+      const validDate = validator.isISO8601(input)
+        && validator.isBefore(input, currentDate);
```

Comment:
`'date'` is generic, but `validator.isBefore` is a restriction that makes using `'date'` not make much sense.

Enforcing `isBefore` belongs in the date picker widget, where it disables the ability to choose any date after "today", but I'm not sure if the widget supports it.

Reply:
Theoretically, but there's a bug with it. https://github.com/pushtell/react-bootstrap-date-picker/issues/139